### PR TITLE
Only run `make release` with tags

### PR DESCRIPTION
**Pull request by dhaiducek**
ref: https://issues.redhat.com/browse/ACM-9106

Followup to:
- #45

### CODE REVIEW

**Comment by dhaiducek** [comment]
This PR got bigger than I expected, but CI is passing and it should be good to go for review now!

File: .github/workflows/olm_tests.yaml [hunk 1]
```diff
@@ -41,20 +41,15 @@ jobs:
       run: |
         # TODO: Use operator-sdk run bundle and bundle-upgrade commands for
         # testing with OLM.
-        REGISTRY_PORT=5000
-        REPO=localhost:${REGISTRY_PORT}
-        IMG=${REPO}/gatekeeper-operator:${GITHUB_SHA}
-        PREV_BUNDLE_INDEX_IMG=quay.io/gatekeeper/gatekeeper-operator-bundle-index:v$(cat REPLACES_VERSION)
```
Comment (dhaiducek) [inline .github/workflows/olm_tests.yaml:47]: (The `PREV_BUNDLE_INDEX_IMG` was removed because these are the default values in the Makefile)

**Review by yiraeChristineKim — APPROVED** [review]

**Comment by yiraeChristineKim** [comment]
Thanks Dale!!

**Pull request merged**